client/consensus/grandpa/authorities.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// AuthorityList A list of Grandpa authorities with associated weights.
+// TODO migrate this type and associated functions to this package
+type AuthorityList []types.Authority
+
+// PendingChange A pending change to the authority set.
+//
+// This will be applied when the announcing block is at some depth within
+// the finalized or unfinalized chain.
+type PendingChange struct {
+	nextAuthorities AuthorityList
+	delay           uint
+	canonHeight     uint
+	canonHash       common.Hash
+	delayKind       DelayKind
+}
+
+// EffectiveNumber Returns the effective number this change will be applied at.
+func (pc *PendingChange) EffectiveNumber() uint {
+	return pc.canonHeight + pc.delay
+}
+
+// AuthoritySetChanges Tracks historical authority set changes. We store the block numbers for the last block
+// of each authority set, once they have been finalized. These blocks are guaranteed to
+// have a justification unless they were triggered by a forced change.
+type AuthoritySetChanges []struct {
+	setId       uint64
+	blockNumber uint
+}
```

Comment:
no receiver methods for this type?  It looks like there are `impl` blocks for this type in substrate ([link](https://github.com/paritytech/substrate/blob/7c8b89b3a6b603902709fc33fdb10a864b8b4204/client/consensus/grandpa/src/authorities.rs#L717))